keps/sig-node/2033-rootless/README.md

```diff
@@ -0,0 +1,806 @@
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1: Production cluster
+
+A user is scared of the past vulnerabilities of kubelet/CRI/OCI, and looking for a way to
+mitigate such potential vulnerabilities.
+
+So the user would want this KEP to be implemented.
+
+The user may face difficulties for deploying stateful workloads because block-based and NFS-based persistent volumes
+mostly do not work (see [Notes/Constraints/Caveats](#notesconstraintscaveats)), but this is not a huge deal,
+when the user can use managed object storages such as Amazon S3, or managed RDBs such as Amazon RDS for storing persistent data.
+
+If the user really needs to run an application that requires the root privileges, the user
+would create a mixed cluster composed of rootful nodes and rootless nodes, and set
+the node selector to ensure the privileged pods to be scheduled on rootful nodes.
+
+#### Story 2: HPC cluster
+
+A user wants to deploy a Kubernetes cluster using shared HPC machines to run scientific research workloads.
+
+However, the machine administrator does not want to allow the user to gain the root privileges,
+because the admin thinks that the user may accidentally break other users' environments.
+
+And yet, the admin hesitates to deploy a shared Kubernetes cluster and to create RBAC-restricted
+accounts for users, because user management in Kubernetes is very difficult.
+
+The user would want this KEP to be implemented so that he/she can deploy Kubernetes without
+convincing the admin.
+
+#### Story 3: `kind` with Rootless Docker/Podman
+
+A user wants to run a test cluster inside Docker/Podman on his/her laptop using `kind`.
+
+However, the user doesn't want Kubernetes/kind/Docker/Podman to gain the root privileges
+because these components may accidentally break the host environment,
+e.g. Docker may modify the host iptables in an unexpected way and break the user's VPN connectivity.
+
+The user would want this KEP to be implemented so that he/she could run `kind` with Rootless Docker/Podman,
+which won't break the host.
+
+### Notes/Constraints/Caveats
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+Privileges:
+- A container with `securityContext.privileged` cannot gain the real root privileges, obviously.
+
+cgroup:
```

Comment:
can you comment on devices?  pids?  swap?  the implication here is the runtime itself is also running rootless too, right?  are there aspects of the host that kubelet monitors today that it cannot monitor?  any exploration of cAdvisor?

Reply:
> can you comment on devices? pids? swap?

Device controller is not supported for rootless, but it's not a huge deal, as non-root user cannot access most devices anyway. PID controller and memory controller (swap) are supported, as well as CPU and IO controller.

> the implication here is the runtime itself is also running rootless too, right? 

Yes

> are there aspects of the host that kubelet monitors today that it cannot monitor? any exploration of cAdvisor?

Most stuff would work, but we may need to modify cAdvisor to monitor non-standard cgroup path (if cAdvisor doesn't support it yet)